src/config.cpp

```diff
@@ -483,12 +504,14 @@ CParseResult CConfig::parseVariable(const std::string& lhs, const std::string& r
 CParseResult CConfig::parseLine(std::string line, bool dynamic) {
     CParseResult result;
 
-    auto         commentPos  = line.find('#');
-    size_t       lastHashPos = 0;
+    line = removeBeginSpacesTabs(line);
+
+    auto   commentPos  = line.find('#');
+    size_t lastHashPos = 0;
 
     while (commentPos != std::string::npos) {
         bool escaped = false;
-        if (commentPos < line.length() - 1) {
+        if (commentPos < line.length() - 1 && commentPos != 0) {
```

Comment:
what's the point of everything but this

Reply:
Line 507 removes leading spaces and tabs so when you hit `#` it is understood as a comment because it can't be keyword, value or section

Reply:
And I thought that the best way for checking if `##` is a start of a comment is to check for first non-space, non-tab character since there is no reason for not having spaces and tabs before the actual comment.

```
    ## Why not allow this for a comment for example
```

Without the other stuff parser would grab `##`, escape it (because `commentPos` won't be 0 in this case) and then tried to parse it as a key, value which would resolve in an error.

Reply:
my question is why the new functions.

Reply:
For better readability I guess, the functionality might be needed in the future and also I thought that removing any spaces and tabs at the end should be done after the comment is removed because if there are any spaces at the end of the _line_ they are part of the actual comment and get removed nevertheless.

They aren't necessary... should I remove them and replace calling them with just `removeBeginEndSpacesTabs`?

Reply:
Done